apps/cli/src/domain/node.ts

```diff
@@ -0,0 +1,37 @@
+import { z } from "zod";
+
+import { Dependencies } from "./dependencies.js";
+import { unwrapOrLogError } from "./index.js";
+
+const ScriptName = z.string().brand<"ScriptName">();
+
+export const ScriptSchema = z.object({
+  name: ScriptName,
+  script: z.string(),
+});
+
+export type Script = z.infer<typeof ScriptSchema>;
+
+export interface NodeReader {
+  getScripts(cwd: string): Promise<Script[]>;
+}
+
+const requiredRootScripts = ["code-review"];
+
+export const checkMonorepoScripts =
+  (monorepoDir: string) =>
+  async (dependencies: Pick<Dependencies, "logger" | "nodeReader">) => {
+    const { logger, nodeReader } = dependencies;
+
+    const scripts = await unwrapOrLogError(dependencies)(() =>
+      nodeReader.getScripts(monorepoDir),
```

Comment:
`unwrapOrLogError` throws, but is not explicit - without looking its code I don't understand what is happening here

I suggest to use https://github.com/supermacro/neverthrow to make error more explicit (and track them at type level)

Reply:
what about using fp-ts instead? I see we need DI too (`make` methods) and `Readers` can help.

Reply:
Honestly, I don't know.
I wouldn't mind keeping it simple without adding the `fp-ts` code; this project can also be maintained by someone approaching TypeScript as a beginner. With `fp-ts`, this is more complex to accomplish